index.js

```diff
@@ -309,8 +309,46 @@ class LedgerBridgeKeyring extends EventEmitter {
     return hdPath
   }
 
-  signTypedData () {
-    throw new Error('Not supported on this device')
+  signTypedData (withAccount, data, options) {
+    const isV4 = options.version === 'V4'
+    const sanitizedData = sigUtil.TypedDataUtils.sanitizeData(data)
+    const domainSeparatorHex = sigUtil.TypedDataUtils.hashStruct('EIP712Domain', sanitizedData.domain, sanitizedData.types, isV4).toString('hex')
+    const hashStructMessageHex = sigUtil.TypedDataUtils.hashStruct(sanitizedData.primaryType, sanitizedData.message, sanitizedData.types, isV4).toString('hex')
+
+    return new Promise((resolve, reject) => {
```

Comment:
Nit: could we make this function `async`, rather than using a Promise constructor here? This is an anti-pattern that we should avoid if at all possible, now that our supported browsers/Node.js versions all support async/await

Reply:
It's not just a superficial concern though - functionally this pattern is dangerous because it lets errors escape uncaught.

For example, if `recoverTypedSignature_v4` throws below on line 342, I believe it would result in an uncaught error. We could carefully comb through this line-by-line to catch all such cases, and wrap them in a `try...catch` to ensure they're captured, but using `async/await` would be simpler and safer.